subsys/bluetooth/mesh/pb_adv.c

```diff
@@ -447,8 +459,20 @@ static void gen_prov_start(struct prov_rx *rx, struct net_buf_simple *buf)
 
 	prov_clear_tx();
 
-	link.rx.seg = (1 << (START_LAST_SEG(rx->gpc) + 1)) - 1;
 	link.rx.last_seg = START_LAST_SEG(rx->gpc);
+
+	/* Confirm that none of the received continuation segments are out of bounds: */
+	if ((link.rx.seg & BIT(0)) && (find_msb_set(~link.rx.seg) >= link.rx.last_seg) {
```

Comment:
I'm afraid I made some compiler errors for you in my snippet. There's a missing closing parenthesis here